addons/web/static/tests/helpers/legacy.js

```diff
@@ -5,15 +5,23 @@ odoo.define("web.SessionOverrideForTests", (require) => {
     // So if a test does a session_reload, it will merge the odoo global of that test
     // into the session, and will alter every subsequent test of the suite.
     // Obviously, we don't want that, ever.
+    const { session: sessionInfo } = require("@web/session");
     const initialOdoo = Object.assign({}, odoo);
     const Session = require("web.Session");
     const { patch } = require("@web/core/utils/patch");
     patch(Session.prototype, "web.SessionTestPatch", {
         async session_reload() {
             const oldOdoo = odoo;
+            const oldSessionInfo = Object.assign({}, sessionInfo);
```

Comment:
I think you'd have to store the sessionInfo during the initial reload (during the session_bind)
and reassign the sessionInfo module to that initial value before calling super.

Here you just delete every key from the module and put them back after